engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala

```diff
@@ -18,8 +18,8 @@ class JobStoreReaderActor(database: JobStoreDatabase) extends Actor with ActorLo
     case QueryJobCompletion(key) =>
       val replyTo = sender()
       database.readJobResult(key) onComplete {
-        case Success(Some(result)) => replyTo ! JobComplete(key, result)
-        case Success(None) => replyTo ! JobNotComplete(key)
+        case Success(Some(result)) => replyTo ! JobComplete(result)
+        case Success(None) => replyTo ! JobNotComplete
         case Failure(t) => replyTo ! JobStoreReadFailure(key, t)
```

Comment:
Mini-shrug but this one still returns the key. 

Also more importantly, it appears to be unhandled.